packages/snaps-rpc-methods/src/permitted/scheduleBackgroundEvent.ts

```diff
@@ -0,0 +1,154 @@
+};
+
+type ScheduleBackgroundEventHookParams = {
+  date: string;
+  request: CronjobRpcRequest;
+};
+
+export type ScheduleBackgroundEventMethodHooks = {
+  scheduleBackgroundEvent: (
+    snapEvent: ScheduleBackgroundEventHookParams,
+  ) => string;
+
+  hasPermission: (permissionName: string) => boolean;
+};
+
+export const scheduleBackgroundEventHandler: PermittedHandlerExport<
+  ScheduleBackgroundEventMethodHooks,
+  ScheduleBackgroundEventParameters,
+  ScheduleBackgroundEventResult
+> = {
+  methodNames: [methodName],
+  implementation: getScheduleBackgroundEventImplementation,
+  hookNames,
+};
+
+const ScheduleBackgroundEventsParametersStruct = object({
+  date: refine(string(), 'date', (val) => {
+    const date = DateTime.fromISO(val);
+    if (date.isValid) {
+      // luxon doesn't have a reliable way to check if timezone info was not provided
+      const count = val.split('').filter((char) => char === '-').length;
+      if (count !== 3 && !val.includes('+') && !val.endsWith('Z')) {
+        return 'ISO8601 string must have timezone information';
+      }
+      return true;
+    }
+    return 'Not a valid ISO8601 string';
+  }),
+  request: CronjobRpcRequestStruct,
+});
+
+export type ScheduleBackgroundEventParameters = InferMatching<
+  typeof ScheduleBackgroundEventsParametersStruct,
+  ScheduleBackgroundEventParams
+>;
+
+/**
+ * The `snap_scheduleBackgroundEvent` method implementation.
+ *
+ * @param req - The JSON-RPC request object.
+ * @param res - The JSON-RPC response object.
+ * @param _next - The `json-rpc-engine` "next" callback. Not used by this
+ * function.
+ * @param end - The `json-rpc-engine` "end" callback.
+ * @param hooks - The RPC method hooks.
+ * @param hooks.scheduleBackgroundEvent - The function to schedule a background event.
+ * @param hooks.hasPermission - The function to check if a snap has the `endowment:cronjob` permission.
+ * @returns An id representing the background event.
+ */
+async function getScheduleBackgroundEventImplementation(
+  req: JsonRpcRequest<ScheduleBackgroundEventParameters>,
+  res: PendingJsonRpcResponse<ScheduleBackgroundEventResult>,
+  _next: unknown,
+  end: JsonRpcEngineEndCallback,
+  {
+    scheduleBackgroundEvent,
+    hasPermission,
+  }: ScheduleBackgroundEventMethodHooks,
+): Promise<void> {
+  const { params, origin } = req as JsonRpcRequest & { origin: string };
+
+  if (!hasPermission(SnapEndowments.Cronjob)) {
+    return end(
+      rpcErrors.invalidRequest({
+        message: `The snap "${origin}" does not have the "${SnapEndowments.Cronjob}" permission.`,
+      }),
+    );
+  }
+
+  try {
+    const validatedParams = getValidatedParams(params);
+
+    const { date, request } = validatedParams;
+
+    // make sure any second/millisecond precision is removed.
+    const truncatedDate = DateTime.fromISO(date, { setZone: true })
+      .startOf('minute')
+      .toISO({
+        suppressMilliseconds: true,
+        suppressSeconds: true,
```

Comment:
Hmm, I'm not sure whether I agree with the SIP that we should truncate to minutes actually.

For instance, if you are trying to schedule something to query each block of a blockchain, 1 minute might be too slow.

Reply:
so just suppressing milliseconds is fine?

Reply:
I think so!